### ISSUE TITLE
Usage with React Spring

### ISSUE THREAD

**Issue by sanbornhilland** [inlet/react-pixi#244]
### Description
I'm wondering if anyone has done any work on using `react-spring` with `react-pixi`. I've been really happy using `react-pixi` but I'm not looking to add some complex animations and leveraging `react-spring` would be very useful. It's been easy so far to use the old renderprops style from `react-spring` but there are a few reasons one might want to use the newer `react-spring` hooks. 

	1. There are some things for which the hooks style is just easier. Some complex animations I'm doing using `Spring.Keyframes` I suspect would be a lot easier with their `hooks` API
	1. The hooks implementation promises better performance. Whether there's a big performance gain when used with `react-pixi` remains to be seen.

Unfortunately as far as I can tell the hooks are only available by wrapping components with the `react-spring` `animated` decorator, the implementation of which is platform specific. They do have a number of targets, including `react-three-fiber`: https://github.com/react-spring/react-spring/blob/master/src/targets/three/index.ts so that may be a useful reference if there was a desire to do create an integration for `react-pixi`.

I'm not intimately familiar with the `react-pixi` or `react-spring` codebases so if anyone has tried this or has more experience with either projects and can provide some input that would be fantastic.

**Comment by inlet** [comment]
Hi @sanbornhilland,

I haven’t create an integration for react spring yet, feel free to create a PR and we collab on it together. 

I’m not familiar with the `react-spring` codebase either, let’s hope integrating this is straight forward.

**Comment by sanbornhilland** [comment]
> Hi @sanbornhilland,
> 
> I haven’t create an integration for react spring yet, feel free to create a PR and we collab on it together.
> 
> I’m not familiar with the `react-spring` codebase either, let’s hope integrating this is straight forward.

Okay. I just briefly started looking into it and surprisingly my initial quick attempt at copying threejs stuff didn't just work :p. I'm likely going to move slow on this because I'm juggling a number of competing concerns at the moment but if I make any progress on it I'll make a PR and would be happy to collaborate.

**Comment by sanbornhilland** [comment]
BTW, @EloB I found [this thread](https://github.com/react-spring/react-spring/issues/901#issue-550315448) on the `react-spring` repo and I'm wondering if you got anywhere with the advice they provided, particularly on following the `konva` target: https://github.com/react-spring/react-spring/blob/v9/targets/konva/src/index.ts

It looks like we have similar aims so if you have started some of this work and would be open to collaborating, that would be great.

**Comment by inlet** [comment]
I just created an integration for react-spring based on the konva target. Expecting to release it somewhere next week.

I think of exporting it as follow, what do you think @sanbornhilland:

```jsx
import { Stage } from '@inlet/react-pixi';
import { Sprite } from '@inlet/react-pixi/animated';
import { Spring } from 'react-spring';

const App = () => (
  <Stage>
    <Spring native to={{ x: 500, y: 500 }}>
      {props => (
        <Sprite image="./image.png" {...props} />
      )}
    </Spring>
  </Stage>
);

```

**Comment by sanbornhilland** [comment]
Wow, you turned that around quickly 👍 

I think the second option you suggested would be better for two reasons.

1. It matches what `react-spring` does so it's familiar to users
1. It clearly separates the `animated.Sprite` from the existing `Sprite`. I think that's useful because you may be using both in the same context and so it's just easier to avoid name collisions if they're namespaced like that.

**Comment by inlet** [comment]
Thanks! the second option it is :smile:

**Comment by inlet** [comment]
PR 👉   #246 

As `react-spring` impacts the import cost quite a bit (~25kb), I opted for:

```jsx
import { Sprite } from '@inlet/react-pixi/animated';
```

It's not merged yet with master, thus available from `@latest`.. but you can try and play around with it by installing the npm dist-tag `react-spring`:

```
npm install @inlet/react-pixi@react-spring
```

**Comment by inlet** [comment]
Fixed in #246
